File: service/src/main/java/com/javaclasses/chatroom/services/DTO/ChatroomDTO.java
package com.javaclasses.chatroom.services.DTO;

import com.javaclasses.chatroom.persistence.entity.Message;
import com.javaclasses.chatroom.persistence.entity.User;

import java.util.List;

public class ChatroomDTO {
    private Long id;
    private String name;
    private List<User> members;
    private List<Message> messages;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<User> getMembers() {
        return members;
    }

    public void setMembers(List<User> members) {
        this.members = members;
    }

    public List<Message> getMessages() {
        return messages;
    }

    public void setMessages(List<Message> messages) {
        this.messages = messages;
    }
}

File: client/src/main/webapp/js/dto/chatroom-dto.js
var ChatroomDTO = function (id, name, creationDate, guests, messages) {

    var getId = function () {
        return id;
    };

    var getName = function () {
        return name;
    };

    var getCreationDate = function () {
        return creationDate;
    };

    var getGuests = function () {
        return guests;
    };

    var getMessages = function () {
        return messages;
    };

    return {
        getId: getId,
        getName: getName,
        getCreationDate: getCreationDate,
        getGuests: getGuests,
        getMessages: getMessages
    };

};